doc(loading): Updated the Usage & Style tab (#4554)
carbon-design-system/carbon-website: src/pages/components/loading/usage.mdx

## Overview

> The Loading component provides visual feedback to indicate that a process or

Reviewer: ```suggestion
The loading component provides visual feedback indicating a process or
```

action is in progress. It helps set users' expectations during wait times by
signaling that the system is working behind the scenes. Depending on the
context, it can be used as a full-page overlay or placed within a specific
section or UI element. Use a loading indicator if the expected wait time exceeds
three seconds.

![Example of the loading component in a UI](images/loading-usage-overview-in-context.png)

<Caption>Example of the loading component in a UI</Caption>

### When to use

- Use when retrieving data or performing slow computations to indicate that a
  request is being processed.
- Use when the entire page or a key section is processing, such as after a user
  submits or saves data.

### When not to use

> - Avoid using loading indicators for progressively displaying content; skeleton

Reviewer: ```suggestion
- Avoid using loading indicators for progressively displaying content; use skeleton
```

#### Large size

The large loading indicator is most commonly used to display loading in
full-screen takeover situations. It can also appear on sections within a page,
in [modals](/components/modal/usage/#loading), and on individual
[tiles](/components/tile/usage). A background overlay is typically placed behind
the large loading indicator so the user cannot interact with the elements
currently in the loading state.

<Row>

<Column colLg={8}>
<GifPlayer color='light'>

![Example of the large loading component in a UI](/images/loading-usage-content-large.gif)

![Example of the large loading component in a UI](/images/loading-usage-content-large.png)

</GifPlayer>
</Column>
</Row>

<Caption>Example of the large loading component in a UI</Caption>

#### Small size

The small loading indicator is best suited for contextual or localized loading
experiences within a specific UI element, particularly in inline loading
scenarios. It works well for quick-loading tasks and can be placed within or
near the triggering element—such as a
[button](/components/button/usage/#loading)—to indicate that a process is
underway. Common examples include saving after clicking on a button, uploading a

> file, or placing the indicator beside a label text to communicate the loading

Reviewer: ```suggestion
file, or placing the indicator beside label text to communicate the loading
```

status. For detailed guidance, refer to the inline loading component
[Usage](/components/inline-loading/usage/) tab.

<Row>

<Column colLg={8}>
<GifPlayer color='light'>

![Example of the small loading component in a UI](/images/modal-error.gif)

![Example of the small loading component in a UI](/images/modal-error.png)

</GifPlayer>
</Column>
</Row>

<Caption>Example of the small loading component in a UI</Caption>

### Placement

#### Large loading placement

The large loading indicator is commonly used for full-screen and component-level
loading scenarios.

> For full-screen loading, place the indicator at the center of the viewport and

Reviewer: I feel like this sentence could be moved into the paragraph above to make this section less choppy.

<Caption>
  Example of the large loading indicator placement for component-level loading
  with an overlay
</Caption>

#### Small loading placement

The small loading indicator is ideal for inline or localized loading within
elements like buttons or compact content areas. It should appear inline with or
adjacent to the related action without using overlays. During the loading state,
any related action buttons—such as “Cancel”—should be temporarily disabled. For
more information about inline loading, refer to the inline loading component
[Usage](/components/inline-loading/usage/) tab.

![Example of the small loading indicator placement for inline loading](images/loading-usage-placement-small-inline.png)

<Caption>
  Example of the small loading indicator placement for inline loading
</Caption>

## Content

### Main elements

The loading indicator can be paired with an optional label to provide additional
context during a loading state.

For large loading indicators, adding a brief status message below the indicator,
such as “Loading data…” can help set users’ expectations. The label text is not
included by default in the component but can be customized to include it if
needed.

> ![Example of the large loading indicator with label text](images/loading-usage-content-large.png)

Reviewer: This image is not showing the label text under the loading indicator.

<Tabs>

<Tab label="Large">

![Large loading component anatomy](images/loading-usage-anatomy-large.png)

</Tab>

<Tab label="Small">

![Small loading component anatomy](images/loading-usage-anatomy-small.png)

</Tab>

</Tabs>

</Column>
</Row>

**Loading indicator:** A circular animated icon indicating that a process is
underway. It serves as the primary visual cue for the loading state.

### Sizing

The Loading component is available in two sizes: **large** and **small**. The
large loading indicator size is the default size.

<Row>
<Column colLg={8}>

![Large and small sizes of loading component](images/loading-usage-sizing.png)

</Column>
</Row>

<Caption>Large and small sizes of loading component</Caption>

#### Large size

The large loading indicator is most commonly used to display loading in
full-screen takeover situations. It can also appear on sections within a page,
in [modals](/components/tile/usage/#loading), and on individual
[tiles](/components/tile/usage). A background overlay is typically placed behind
the large loading indicator so the user can not interact with the elements
currently in the loading state.

<Row>

<Column colLg={8}>
<GifPlayer color='light'>

> ![Example of the small loading component in a UI](/images/loading-usage-content-large.gif)

Author: Hi Lauren! I’ll check with Thy or a developer to confirm if this is coming from the code, as I haven’t added anything to the image myself. Aside from that, I’ve addressed all the other feedback. Thank you! ✨

experiences within a specific UI element, particularly in inline loading
scenarios. It works well for quick-loading tasks and can be placed within or
near the triggering element—such as a
[button](/components/button/usage/#loading)—to indicate that a process is
underway. Common examples include saving after a button click, uploading a file,
or placing the indicator beside a text label to communicate loading status. For
detailed guidance, refer to the inline loading component
[Usage](/components/inline-loading/usage/) tab.

<Row>

<Column colLg={8}>
<GifPlayer color='light'>

![Example of the large loading component in a UI](/images/modal-error.gif)

![Example of the large loading component in a UI](/images/modal-error.png)

</GifPlayer>
</Column>
</Row>

<Caption>Example of the small loading component in a UI</Caption>

### Placement

#### Large loading placement

The large loading indicator is commonly used for full-screen and component-level
loading scenarios.

For full-screen loading, place the indicator at the center of the viewport and
apply a semi-transparent overlay to signal that the entire interface is
inactive.

<Row>

> <Column colLg={8}>

Reviewer: ```suggestion
<Column colLg={12}>
```

> ![Example of large loading indicator placement for full-screen loading with overlay](images/loading-usage-placement-large-full-screen.png)

Reviewer: ```suggestion
![Example of the large loading indicator placement for full-screen loading with an overlay](images/loading-usage-placement-large-full-screen.png)
```

</Column>
</Row>

<Caption>

>   Example of large loading indicator placement for full-screen loading with

Reviewer: ```suggestion
  Example of the large loading indicator placement for full-screen loading with an
```

  overlay
</Caption>

> When used within a component—such as a modal, tile, or side panel—center the

Reviewer: ```suggestion
When used within a component, such as a modal, tile, or side panel, center the
```

indicator in that space and use an overlay to prevent interaction with the
content while it is loading.

<Row>
<Column colLg={8}>

> ![Example of large loading indicator placement for component-level loading with overlay](images/loading-usage-placement-large-component-level.png)

Reviewer: ```suggestion
![Example of the large loading indicator placement for component-level loading with an overlay](images/loading-usage-placement-large-component-level.png)
```

</Column>
</Row>

<Caption>

>   Example of large loading indicator placement for component-level loading with

Reviewer: ```suggestion
  Example of the large loading indicator placement for component-level loading with an
```

  overlay
</Caption>

#### Small loading placement

The small loading indicator is ideal for inline or localized loading within
elements like buttons or compact content areas. It should appear inline with or

> adjacent to the related action, without using overlays. During the loading

Reviewer: ```suggestion
adjacent to the related action without using overlays. During the loading
```

state, any related action buttons—such as “Cancel”—should be disabled. For more
information about inline loading, refer to the inline loading component
[Usage](/components/inline-loading/usage/) tab.

<Row>

> <Column colLg={8}>

Reviewer: ```suggestion
<Column colLg={12}>
```

> ![Example of small loading indicator placement for inline loading](images/loading-usage-placement-small-inline.png)

Reviewer: ```suggestion
![Example of the small loading indicator placement for inline loading](images/loading-usage-placement-small-inline.png)
```

</Column>
</Row>

<Caption>

>   Example of small loading indicator placement for inline loading

Reviewer: ```suggestion
  Example of the small loading indicator placement for inline loading
```

</Caption>

## Content

### Main elements

> The loading indicator can be paired with optional text to provide additional

Reviewer: ```suggestion
The loading indicator can be paired with optional label text to provide additional
```

context during a loading state.

> For **large loading** indicators, adding a brief status message below the

Reviewer: I am going to suggest rewording this sentence as the following. You do not need to semibold in this sentence:

For large loading indicators, adding a brief status message below the indicator, such as “Loading data…” can help set users’ expectations. The label text is not included by default in the component but can be customized to include it if needed.

indicator as “Loading data…”—can help set expectations for users. This text is
not included by default in the component and should be added manually if needed.

> ![Example of large loading indicator with label](images/loading-usage-content-large.png)

Reviewer: ```suggestion
![Example of the large loading indicator with label text](images/loading-usage-content-large.png)
```

> <Caption>Example of large loading indicator with label</Caption>

Reviewer: ```suggestion
<Caption>Example of the large loading indicator with label text</Caption>
```

> For **small loading** indicators, especially in inline use, content should be

Reviewer: ```suggestion
For small loading indicators, especially in inline use, content should be
```

concise and placed near the related action. For more detailed guidance, refer to
the inline loading component’s
[Content](/components/inline-loading/usage/#content) section.

### Further guidance

For further content guidance, see Carbon's
[content guidelines](/guidelines/content/overview).

## Behaviors

### States

> There are two states to the loading component: **inactive**, and **active**. For

Reviewer: ```suggestion
There are two states of the loading component: **inactive** and **active**. For
```

detailed visual information about the various states for this component, see the
[Style](/components/loading/style/) tab.

| State    | When to use                                                                             |
| -------- | --------------------------------------------------------------------------------------- |

> | Inactive | The inactive loading state is when no data is being loaded, it has no visual indicator. |

Reviewer: ```suggestion
| Inactive | The inactive loading state is when no data is being loaded and has no visual indicator. |
```

| Active   | The active loading state indicates that the action is still in progress.                |

> <Row>

Reviewer: This image needs to be removed.

<Column colLg={8}>

<Tabs>

<Tab label="Large">

![Inactive and active states of large loading component](images/loading-usage-states-large.png)

</Tab>

<Tab label="Small">

![Inactive and active states of small loading component](images/loading-usage-states-small.png)

</Tab>

</Tabs>

</Column>
</Row>

### Interactions

For large loading indicators, a semi-transparent overlay is applied over the
page or component, blocking all interaction. During this state, users should not
be able to interact with any elements, as the overlay indicates the interface is
temporarily unavailable.

> For small loading indicators used in inline loading scenarios, interactive

Reviewer: ```suggestion
For small loading indicators in inline loading scenarios, interactive
```

elements like buttons should be temporarily disabled while loading is in
progress. This prevents repeated actions and ensures a smooth user experience.

> Learn more in the inline loading –

Reviewer: ```suggestion
For more guidance, see the inline loading component 
```
